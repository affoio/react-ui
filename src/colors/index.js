const colors = {
  indigoBlue: {
    800: '#4718b4',
    700: '#623abf',
    600: '#8769ce',
    500: '#ac97dd',
    400: '#d1c5ec',
    200: '#ece7f7',
    300: '#e4ddf4',
    100: '#f6f3fb',
  },
  marigold: {
    800: '#ffca00',
    700: '#ffd226',
    600: '#ffdc59',
    500: '#ffe78c',
    400: '#fff2bf',
    300: '#fff7d9',
    200: '#fff9e5',
    100: '#fffcf2',
  },
  cerise: {
    800: '#e2086e',
    700: '#e62d84',
    600: '#ec5ea1',
    500: '#f290be',
    400: '#f8c1db',
    300: '#fbdaea',
    200: '#fce6f0',
    100: '#fdf2f8',
  },
  black: {
    800: '#000000',
    700: '#262626',
    600: '#595959',
    500: '#8c8c8c',
    400: '#bfbfbf',
    300: '#d9d9d9',
    200: '#e5e5e5',
    100: '#f2f2f2',
  },
  seaweed: {
    800: '#13c64f',
    700: '#36ce69',
    600: '#65da8c',
    500: '#95e5b0',
    400: '#c4f1d3',
    300: '#dcf7e5',
    200: '#e7f9ed',
    100: '#f3fcf6',
  },
  white: '#ffffff',
};

export default colors;
