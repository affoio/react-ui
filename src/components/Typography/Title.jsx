import React from 'react';
import { oneOf, bool } from 'prop-types';

import Text from './Text';

const sizeFromLevel = {
  '1': '600',
  '2': '500',
  '3': '300',
};

export const propTypes = {
  level: oneOf(['1', '2', '3']),
  inverse: bool,
};

const defaultProps = {
  level: '1',
};

const Title = ({ children, level, ...otherProps }) => {
  const size = sizeFromLevel[level];

  return (
    <Text {...otherProps} size={size} fontWeight="bold">
      {children}
    </Text>
  );
};

Title.propTypes = propTypes;
Title.defaultProps = defaultProps;

export default Title;
