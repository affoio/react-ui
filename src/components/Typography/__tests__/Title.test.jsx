import React from 'react';
import { render } from 'enzyme';
import renderer from 'react-test-renderer';
import Title from '../Title';
import colors from '../../../colors';

describe('<Title>', () => {
  it('renders correctly', () => {
    const tree = render(<Title>Lorem ipsum, dolor sit amet consectetur adipisicing elit. Quaerat, voluptas.</Title>);
    expect(tree).toMatchSnapshot();
  });

  it('render inversed Title', () => {
    const tree = renderer
      .create(<Title inverse>Lorem ipsum, dolor sit amet consectetur adipisicing elit. Quaerat, voluptas.</Title>)
      .toJSON();

    expect(tree).toHaveStyleRule(`color: ${colors.white}`);
  });

  describe('renders with different levels', () => {
    const levels = ['1', '2', '3'];

    levels.forEach(level => {
      it(`render Title with size=${level}`, () => {
        const tree = render(
          <Title level={level}>Lorem ipsum, dolor sit amet consectetur adipisicing elit. Quaerat, voluptas.</Title>
        );
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
