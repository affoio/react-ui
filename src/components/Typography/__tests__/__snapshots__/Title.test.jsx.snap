// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title> renders correctly 1`] = `
.c0 {
  display: inline;
  font-weight: bold;
  font-size: 32px;
  line-height: 44px;
  color: inherit;
}

<div
  class="c0"
>
  Lorem ipsum, dolor sit amet consectetur adipisicing elit. Quaerat, voluptas.
</div>
`;

exports[`<Title> renders with different levels render Title with size=1 1`] = `
.c0 {
  display: inline;
  font-weight: bold;
  font-size: 32px;
  line-height: 44px;
  color: inherit;
}

<div
  class="c0"
>
  Lorem ipsum, dolor sit amet consectetur adipisicing elit. Quaerat, voluptas.
</div>
`;

exports[`<Title> renders with different levels render Title with size=2 1`] = `
.c0 {
  display: inline;
  font-weight: bold;
  font-size: 24px;
  line-height: 32px;
  color: inherit;
}

<div
  class="c0"
>
  Lorem ipsum, dolor sit amet consectetur adipisicing elit. Quaerat, voluptas.
</div>
`;

exports[`<Title> renders with different levels render Title with size=3 1`] = `
.c0 {
  display: inline;
  font-weight: bold;
  font-size: 16px;
  line-height: 25px;
  color: inherit;
}

<div
  class="c0"
>
  Lorem ipsum, dolor sit amet consectetur adipisicing elit. Quaerat, voluptas.
</div>
`;
